modules/claims_api/app/controllers/claims_api/application_controller.rb

```diff
@@ -133,5 +133,15 @@ def veteran_from_headers(with_gender: false)
       vet.participant_id = header('X-VA-PID') || vet.mpi.profile&.participant_id
       vet
     end
+
+    def authenticate_token
+      super
+    rescue => e
+      log_message_to_sentry('Authentication Error in claims',
+                            :warning,
+                            body: e.message)
+      render json: { errors: [{ status: 401, detail: 'User not a valid or authorized Veteran for this end point.' }] },
+             status: :unauthorized
+    end
```

Comment:
why not make this change higher up? this seems to make sense for any endpoint within vets-api no?

Reply:
So in this case we are hijacking the authorization method from the Open ID controller and just injecting a raise in the event anything goes wrong with validating the user, All of our controllers inherit from this module controller so made the most sense to put it here so that it'll apply across all of the end points minus the ones in V0 which skip token authentication so this wouldn't apply to them anyhow.

Reply:
> All of our controllers inherit from this module controller

All of our controllers within the claims module correct? I guess I was more asking why this change doesn't make sense higher up so to affect all modules within the api, but maybe thats simply frowned upon? or maybe each module uses authentication slightly differently rendering a broad change like this almost impossible?

Fair enough though :+1:

Reply:
+1 to @SlakrHakr 's question. `authenticate_token` doesn't have any explicit error handling in the `OpenidApplicationController`. I can't see why we wouldn't want this to be higher up.

Put another way: this message feels general enough to live at a higher level of abstraction from claims, but is it too specific to put all the way at the top?

Reply:
Each module can have different authentication rules because they belong to different teams and use cases, what we are inheriting is a simpler implementation that we are adding Respresentative Oauth rules that only we need to support for now. Hopefully one day Veteran Verification or even VA.gov support non veterans and we can move this stuff a bit higher upstream but at the moment we are the only ones with this particular use case.